virttest/qemu_devices/qcontainer.py

```diff
@@ -2121,10 +2118,16 @@ def images_define_by_params(self, name, image_params, media=None,
 
         # add storage access secret object for image
         image_access = storage.retrieve_access_info(name, image_params)
+        image_filename = storage.get_image_filename(image_params, data_root)
+
+        if (Flags.BLOCKDEV in self.caps and
+                image_params.get("image_snapshot") == "yes"):
+            image_filename = qemu_storage.create_image_offline_snapshot(
```

Comment:
Better to leave a log message to users, e.g.
```python
logging.info("'snapshot=on' is not supported by '-blockdev' but requested from the image '%s', imitating the behavior of '-drive' to keep compatibility", name)
```

Reply:
Thanks, updated.